cpp/src/io/parquet/compact_protocol_writer.hpp

```diff
@@ -90,6 +91,8 @@ class CompactProtocolFieldWriter {
   template <typename Enum>
   inline void field_int_list(int field, std::vector<Enum> const& val);
 
+  inline void field_int_list(int field, const std::vector<int64_t>& val);
```

Comment:
Nice catch...some of the "new" code here predates the great east const migration of 2023 :) Will fix.

I'll see if I can get away with the one declaration and keep the int64_t specialization in the source file, since the field type is different (ST_FLD_I64 vs ST_FLD_I32 for the enums).